tests/benchmark/test_osu.py

```diff
@@ -6,6 +6,57 @@
 from .. import FakeCluster
 
 
+class TestOSUOptionsAttributes(unittest.TestCase):
+    def test_categories_assignment(self):
+        benchmark = OSU()
+        # completely replace attributes
+        benchmark.attributes = dict(
+            executable='/path/to/fake',
+            srun_nodes=['node01', 'node03', 'node05'],
+            categories=['osu_bw', 'osu_latency'],
+            options={
+                'osu_bw': '[-x, 600, -i, 400]',
+                'osu_latency': '[-x, 500, -i, 200]',
+            },
+        )
+
+        self.assertEqual(
+            benchmark.arguments,
+            {'osu_bw': '[-x, 600, -i, 400]', 'osu_latency': '[-x, 500, -i, 200]'},
+        )
+
+    def test_partial_override(self):
+        benchmark = OSU()
+        # completely replace attributes
+        benchmark.attributes = dict(
+            executable='/path/to/fake',
+            srun_nodes=['node01', 'node03', 'node05'],
+            categories=['osu_bw', 'osu_latency'],
+            options={'osu_bw': '[-x, 600, -i, 400]'},
```

Comment:
1. `'[-x, 600, -i, 400]'` should be a list not a string.
1. Put it in a local variable to avoid duplication below

Reply:
Not sure what you mean here.

Reply:
1. `options={'osu_bw': '[-x, 600, -i, 400]'},` is wrong. should be `options={'osu_bw': [-x, 600, -i, 400]},` instead.
2.  `[-x, 600, -i, 400]` should not appear twice in the function. Instead you should define a variable

Reply:
That's what I thought it was. But if I do that I get an undefined reference to "x".

Reply:
grrrrrr `['-x', 600, '-i', 400]`

Reply:
Ok, now it makes sense. Sorry for bothering you with such small stuff.

Reply:
Done!